packages/backend-defaults/src/entrypoints/urlReader/lib/ConfluenceUrlReader.ts

```diff
@@ -0,0 +1,273 @@
+/*
+ * Copyright 2024 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {
+  LoggerService,
+  UrlReaderService,
+  UrlReaderServiceReadTreeOptions,
+  UrlReaderServiceReadTreeResponse,
+  UrlReaderServiceReadUrlOptions,
+  UrlReaderServiceReadUrlResponse,
+  UrlReaderServiceSearchOptions,
+  UrlReaderServiceSearchResponse,
+} from '@backstage/backend-plugin-api';
+import {
+  FromReadableArrayOptions,
+  ReaderFactory,
+  ReadTreeResponseFactory,
+} from './types';
+import {
+  Attachment,
+  ChildPage,
+  ConfluenceIntegrationConfig,
+  readConfluenceIntegrationConfigs,
+} from '@backstage/integration';
+import { Readable } from 'stream';
+import { NotFoundError, NotModifiedError } from '@backstage/errors';
+import { NodeHtmlMarkdown } from 'node-html-markdown';
+
+export class ConfluenceUrlReader implements UrlReaderService {
+  static factory: ReaderFactory = ({ config, logger, treeResponseFactory }) => {
+    return readConfluenceIntegrationConfigs(config).map(integration => {
+      const reader = new ConfluenceUrlReader(integration, logger, {
+        treeResponseFactory,
+      });
+      const predicate = (url: URL) => url.host === integration.host;
+      return { reader, predicate };
+    });
+  };
+
+  constructor(
+    private readonly config: ConfluenceIntegrationConfig,
+    private readonly logger: LoggerService,
+    private readonly deps: {
+      treeResponseFactory: ReadTreeResponseFactory;
+    },
+  ) {}
+
+  async readUrl(
+    _url: string,
+    _options?: UrlReaderServiceReadUrlOptions,
+  ): Promise<UrlReaderServiceReadUrlResponse> {
+    throw new Error('This method will be implemented in the future'); //+
+  }
+
+  parseUrl(url: string) {
+    // url format: https://<company-domain>/wiki/spaces/<space-id>/pages/<page-id>/<page-title>
```

Comment:
Can we add support for on-prem? That's going to come up right away if this moved forward. There are some examples here: https://github.com/backstage/backstage/blob/master/plugins/scaffolder-backend-module-confluence-to-markdown/src/actions/confluence/helpers.ts

Reply:
Added support for on-perm @awanlin ..and added docs for confluence integration as well.
Could you please check them both?